src/decode.rs

```diff
@@ -0,0 +1,318 @@
+        .map(|s| s.to_string())
+        .or(config.abi_path.clone())        
+}
+
+pub fn create_decode_command<'a, 'b>() -> App<'a, 'b> {
+    SubCommand::with_name("decode")
+        .about("Decode commands.")
+        .setting(AppSettings::AllowLeadingHyphen)  
+        .setting(AppSettings::TrailingVarArg)
+        .setting(AppSettings::DontCollapseArgsInUsage)
+        .subcommand(SubCommand::with_name("body")
+            .arg(Arg::with_name("BODY")
+                .required(true)
+                .help("Message body encoded as base64."))
+            .arg(Arg::with_name("ABI")
+                .long("--abi")
+                .takes_value(true)
+                .help("Path to ABI file.")))
+        .subcommand(SubCommand::with_name("msg")
+           .arg(Arg::with_name("MSG")
+                    .required(true)
+                    .help("Path to message boc file."))
+            .arg(Arg::with_name("ABI")
+                    .long("--abi")
+                    .takes_value(true)
+                    .help("Path to ABI file.")))
+}
+
+pub fn decode_command(m: &ArgMatches, config: Config) -> Result<(), String> {
+    if let Some(m) = m.subcommand_matches("body") {
+        return decode_body_command(m, config);
+    }
+    if let Some(m) = m.subcommand_matches("msg") {
+        return decode_message_command(m, config);
+    }
+    Err("unknown command".to_owned())
+}
+
+fn decode_body_command(m: &ArgMatches, config: Config) -> Result<(), String> {
+    let body = m.value_of("BODY");
+    let abi = Some(
+        match_abi_path(m, &config)
+            .ok_or("ABI file not defined. Supply it in config file or command line.".to_string())?
+    );
+    if !config.is_json {
+        print_args!(m, body, abi);
+    }
+    println!("{}", decode_body(body.unwrap(), &abi.unwrap(), config.is_json)?);
+    Ok(())
+}
+
+fn decode_message_command(m: &ArgMatches, config: Config) -> Result<(), String> {
+    let msg = m.value_of("MSG");
+    let abi = Some(
+        match_abi_path(m, &config)
+            .ok_or("ABI file not defined. Supply it in config file or command line.".to_string())?
+    );
+    if !config.is_json {
+        print_args!(m, msg, abi);
+    }
+    let msg = msg.map(|f| std::fs::read(f))
+        .transpose()
+        .map_err(|e| format!(" failed to read msg boc file: {}", e))?
+        .unwrap();
+    println!("{}", decode_message(msg, abi, config.is_json)?);
+    Ok(())
+}
+
+fn print_decoded_body(body_vec: Vec<u8>, abi: &str, is_json: bool) -> Result<String, String> {
+    let ton = create_client_local()?;
+    let res = ton.contracts.decode_input_message_body(
+        abi.into(),
+        &body_vec,
+        false,
+    ).or_else(|_| ton.contracts.decode_input_message_body(
+        abi.into(),
+        &body_vec,
+        true,
+    )).or_else(|e| {
+        let mut boc = vec![];
+        serialize_tree_of_cells(&Cell::default(), &mut boc).unwrap();
+        if hex::encode(&boc) == hex::encode(&body_vec) {
```

Comment:
Why do you compare string representation of vectors? I think comparing vectors directly is better